core/src/main/java/nl/nn/adapterframework/parameters/Parameter.java

```diff
@@ -680,13 +694,21 @@ private Object getValueForFormatting(ParameterValueList alreadyResolvedParameter
 		Object substitutionValue = paramValue == null ? null : paramValue.getValue();
 
 		if (substitutionValue == null) {
-			substitutionValue = session.get(name);
+			substitutionValue = session.getMessage(name);
 		}
 		if (substitutionValue instanceof Message) {
-			try {
-				substitutionValue = ((Message)substitutionValue).asString();
-			} catch (IOException e) {
-				throw new ParameterException("Cannot get substitution value", e);
+			Message substitutionValueMessage = (Message)substitutionValue;
+			if (substitutionValueMessage.isEmpty()) {
+				substitutionValue = null;
+			} else if (substitutionValueMessage.asObject() instanceof Date) {
+				SimpleDateFormat formatterFrom = new SimpleDateFormat(PutSystemDateInSession.FORMAT_FIXEDDATETIME);
+				substitutionValue = formatterFrom.format((Date)substitutionValueMessage.asObject());
```

Comment:
dit hoeft toch niet? een date is gewoon een date.

Reply:
maar hier moet het geen Date zijn, omdat het hier een pattern is dat naar een string gerendered wordt.